js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetTagFilter.tsx

```diff
@@ -74,6 +74,23 @@ export function useAssetTagsForAssets(
   );
 }
 
+export function useAssetKindsForAssets(
+  assets: {definition?: {kinds?: string[] | null} | null}[],
+): string[] {
+  return useMemo(
+    () =>
+      Array.from(
+        new Set(
+          assets
+            .map((a) => a?.definition?.kinds)
+            .filter((o) => o)
+            .flatMap((a) => a as string[]),
+        ),
+      ).sort(),
```

Comment:
I think you can just do this:

```
Array.from(
  new Set(assets.map((a) => a?.definition?.kinds || []).flat())
).sort(...);
```

Also, it's generally good to use this to ensure a reliable string sort:

```
.sort((a, b) => COMMON_COLLATOR.compare(a, b))
```